src/Compilers/Core/Portable/Operations/ControlFlowGraph.Region.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    public partial class ControlFlowGraph
+    {
+        /// <summary>
+        /// Defines kinds of regions that can be present in a <see cref="ControlFlowGraph"/>
+        /// </summary>
+        public enum RegionKind
+        {
+            /// <summary>
+            /// A root region encapsulating all <see cref="BasicBlock"/>s in a <see cref="ControlFlowGraph"/>
+            /// </summary>
+            Root,
+
+            /// <summary>
+            /// Region with the only purpose to represent the life-time of locals
```

Comment:
>life-time [](start = 62, length = 9)

We should probably be more precise about what lifetime means here, perhaps with some remarks. Is the lifetime the entire scope that the variable is declared, or is it after declaration?